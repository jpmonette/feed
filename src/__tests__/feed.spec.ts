import Feed from "../feed";

let updated = new Date("Sat, 13 Jul 2013 23:00:00 GMT");
let published = new Date("Sat, 10 Jul 2013 23:00:00 GMT");

let feed = new Feed({
  title: "Feed Title",
  description: "This is my personnal feed!",
  link: "http://example.com/",
  id: "http://example.com/",
  feed: "http://example.com/feed.rss",
  image: "http://example.com/image.png",
  copyright: "All rights reserved 2013, John Doe",
  updated, // optional, default = today

  author: {
    name: "John Doe",
    email: "johndoe@example.com",
    link: "https://example.com/johndoe"
  }
});

feed.addCategory("Technology");

feed.addContributor({
  name: "Johan Cruyff",
  email: "johancruyff@example.com",
  link: "https://example.com/johancruyff"
});

feed.addItem({
  title: "Hello World",
  id: "https://example.com/hello-world",
  link: "https://example.com/hello-world",
  description: "This is an article about Hello World.",
  content: "Content of my item",
  author: [
    {
      name: "Jane Doe",
      email: "janedoe@example.com",
      link: "https://example.com/janedoe"
    },
    {
      name: "Joe Smith",
      email: "joesmith@example.com",
      link: "https://example.com/joesmith"
    }
  ],
  contributor: [
    {
      name: "Shawn Kemp",
      email: "shawnkemp@example.com",
      link: "https://example.com/shawnkemp"
    },
    {
      name: "Reggie Miller",
      email: "reggiemiller@example.com",
      link: "https://example.com/reggiemiller"
    }
  ],
  extensions: [
    {
      name: "_item_extension_1",
      objects: {
        about: "just an item extension example",
        dummy1: "example"
      }
    },
    {
      name: "_item_extension_2",
      objects: {
        about: "just a second item extension example",
        dummy1: "example"
      }
    }
  ],
  date: updated,
  image: "https://example.com/hello-world.jpg",
  published
});

feed.addExtension({
  name: "_example_extension",
  objects: {
    about: "just an extension example",
    dummy: "example"
  }
});

describe("rss 2.0", () => {
  it("should generate a valid feed", () => {
    let actual = feed.rss2();
    expect(actual).toMatchSnapshot();
  });

  it("should generate a valid feed - legacy", () => {
    let actual = feed.render("rss-2.0");
    expect(actual).toMatchSnapshot();
  });
});

describe("atom 1.0", () => {
  it("should generate a valid feed", () => {
    let actual = feed.atom1();
    expect(actual).toMatchSnapshot();
  });

  it("should generate a valid feed - legacy", () => {
    let actual = feed.render("atom-1.0");
    expect(actual).toMatchSnapshot();
  });
});

describe("json 1", () => {
  it("should generate a valid feed", () => {
    let actual = JSON.parse(feed.json1());
    expect(actual).toMatchSnapshot();
  });
});
