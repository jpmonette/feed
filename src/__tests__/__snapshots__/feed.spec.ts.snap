// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atom 1.0 should generate a valid feed - legacy 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<feed xmlns=\\"http://www.w3.org/2005/Atom\\">
    <id>http://example.com/</id>
    <title>Feed Title</title>
    <updated>2013-07-13T23:00:00.000Z</updated>
    <generator>https://github.com/jpmonette/feed</generator>
    <author>
        <name>John Doe</name>
        <email>johndoe@example.com</email>
        <uri>https://example.com/johndoe</uri>
    </author>
    <link rel=\\"alternate\\" href=\\"http://example.com/\\"/>
    <link rel=\\"self\\" href=\\"http://example.com/feed.rss\\"/>
    <subtitle>This is my personnal feed!</subtitle>
    <logo>http://example.com/image.png</logo>
    <rights>All rights reserved 2013, John Doe</rights>
    <category term=\\"Technology\\">
    </category>
    <contributor>
        <name>Johan Cruyff</name>
        <email>johancruyff@example.com</email>
        <uri>https://example.com/johancruyff</uri>
    </contributor>
    <entry>
        <title type=\\"html\\"><![CDATA[Hello World]]></title>
        <id>https://example.com/hello-world</id>
        <link href=\\"https://example.com/hello-world\\">
        </link>
        <updated>2013-07-13T23:00:00.000Z</updated>
        <summary type=\\"html\\"><![CDATA[This is an article about Hello World.]]></summary>
        <content type=\\"html\\"><![CDATA[Content of my item]]></content>
        <author>
            <name>Jane Doe</name>
            <email>janedoe@example.com</email>
            <uri>https://example.com/janedoe</uri>
        </author>
        <author>
            <name>Joe Smith</name>
            <email>joesmith@example.com</email>
            <uri>https://example.com/joesmith</uri>
        </author>
        <contributor>
            <name>Shawn Kemp</name>
            <email>shawnkemp@example.com</email>
            <uri>https://example.com/shawnkemp</uri>
        </contributor>
        <contributor>
            <name>Reggie Miller</name>
            <email>reggiemiller@example.com</email>
            <uri>https://example.com/reggiemiller</uri>
        </contributor>
        <published>2013-07-10T23:00:00.000Z</published>
    </entry>
</feed>"
`;

exports[`atom 1.0 should generate a valid feed 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<feed xmlns=\\"http://www.w3.org/2005/Atom\\">
    <id>http://example.com/</id>
    <title>Feed Title</title>
    <updated>2013-07-13T23:00:00.000Z</updated>
    <generator>https://github.com/jpmonette/feed</generator>
    <author>
        <name>John Doe</name>
        <email>johndoe@example.com</email>
        <uri>https://example.com/johndoe</uri>
    </author>
    <link rel=\\"alternate\\" href=\\"http://example.com/\\"/>
    <link rel=\\"self\\" href=\\"http://example.com/feed.rss\\"/>
    <subtitle>This is my personnal feed!</subtitle>
    <logo>http://example.com/image.png</logo>
    <rights>All rights reserved 2013, John Doe</rights>
    <category term=\\"Technology\\">
    </category>
    <contributor>
        <name>Johan Cruyff</name>
        <email>johancruyff@example.com</email>
        <uri>https://example.com/johancruyff</uri>
    </contributor>
    <entry>
        <title type=\\"html\\"><![CDATA[Hello World]]></title>
        <id>https://example.com/hello-world</id>
        <link href=\\"https://example.com/hello-world\\">
        </link>
        <updated>2013-07-13T23:00:00.000Z</updated>
        <summary type=\\"html\\"><![CDATA[This is an article about Hello World.]]></summary>
        <content type=\\"html\\"><![CDATA[Content of my item]]></content>
        <author>
            <name>Jane Doe</name>
            <email>janedoe@example.com</email>
            <uri>https://example.com/janedoe</uri>
        </author>
        <author>
            <name>Joe Smith</name>
            <email>joesmith@example.com</email>
            <uri>https://example.com/joesmith</uri>
        </author>
        <contributor>
            <name>Shawn Kemp</name>
            <email>shawnkemp@example.com</email>
            <uri>https://example.com/shawnkemp</uri>
        </contributor>
        <contributor>
            <name>Reggie Miller</name>
            <email>reggiemiller@example.com</email>
            <uri>https://example.com/reggiemiller</uri>
        </contributor>
        <published>2013-07-10T23:00:00.000Z</published>
    </entry>
</feed>"
`;

exports[`json 1 should generate a valid feed 1`] = `
Object {
  "_example_extension": Object {
    "about": "just an extension example",
    "dummy": "example",
  },
  "author": Object {
    "name": "John Doe",
    "url": "https://example.com/johndoe",
  },
  "description": "This is my personnal feed!",
  "home_page_url": "http://example.com/",
  "icon": "http://example.com/image.png",
  "items": Array [
    Object {
      "_item_extension_1": Object {
        "about": "just an item extension example",
        "dummy1": "example",
      },
      "_item_extension_2": Object {
        "about": "just a second item extension example",
        "dummy1": "example",
      },
      "author": Object {
        "name": "Jane Doe",
        "url": "https://example.com/janedoe",
      },
      "date_modified": "2013-07-13T23:00:00.000Z",
      "date_published": "2013-07-10T23:00:00.000Z",
      "html_content": "Content of my item",
      "id": "https://example.com/hello-world",
      "image": "https://example.com/hello-world.jpg",
      "summary": "This is an article about Hello World.",
      "title": "Hello World",
      "url": "https://example.com/hello-world",
    },
  ],
  "title": "Feed Title",
  "version": "https://jsonfeed.org/version/1",
}
`;

exports[`rss 2.0 should generate a valid feed - legacy 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<rss version=\\"2.0\\" xmlns:content=\\"http://purl.org/rss/1.0/modules/content/\\" xmlns:atom=\\"http://www.w3.org/2005/Atom\\">
    <channel>
        <title>Feed Title</title>
        <link>http://example.com/</link>
        <description>This is my personnal feed!</description>
        <lastBuildDate>Sat, 13 Jul 2013 23:00:00 GMT</lastBuildDate>
        <docs>http://blogs.law.harvard.edu/tech/rss</docs>
        <generator>https://github.com/jpmonette/feed</generator>
        <image>
            <title>Feed Title</title>
            <url>http://example.com/image.png</url>
            <link>http://example.com/</link>
        </image>
        <copyright>All rights reserved 2013, John Doe</copyright>
        <category>Technology</category>
        <atom:link href=\\"http://example.com/feed.rss\\" rel=\\"self\\" type=\\"application/rss+xml\\"/>
        <item>
            <title><![CDATA[Hello World]]></title>
            <link>https://example.com/hello-world</link>
            <guid>https://example.com/hello-world</guid>
            <pubDate>Sat, 13 Jul 2013 23:00:00 GMT</pubDate>
            <description><![CDATA[This is an article about Hello World.]]></description>
            <content:encoded><![CDATA[Content of my item]]></content:encoded>
            <author>janedoe@example.com (Jane Doe)</author>
            <author>joesmith@example.com (Joe Smith)</author>
            <enclosure url=\\"https://example.com/hello-world.jpg\\">
            </enclosure>
        </item>
    </channel>
</rss>"
`;

exports[`rss 2.0 should generate a valid feed 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<rss version=\\"2.0\\" xmlns:content=\\"http://purl.org/rss/1.0/modules/content/\\" xmlns:atom=\\"http://www.w3.org/2005/Atom\\">
    <channel>
        <title>Feed Title</title>
        <link>http://example.com/</link>
        <description>This is my personnal feed!</description>
        <lastBuildDate>Sat, 13 Jul 2013 23:00:00 GMT</lastBuildDate>
        <docs>http://blogs.law.harvard.edu/tech/rss</docs>
        <generator>https://github.com/jpmonette/feed</generator>
        <image>
            <title>Feed Title</title>
            <url>http://example.com/image.png</url>
            <link>http://example.com/</link>
        </image>
        <copyright>All rights reserved 2013, John Doe</copyright>
        <category>Technology</category>
        <atom:link href=\\"http://example.com/feed.rss\\" rel=\\"self\\" type=\\"application/rss+xml\\"/>
        <item>
            <title><![CDATA[Hello World]]></title>
            <link>https://example.com/hello-world</link>
            <guid>https://example.com/hello-world</guid>
            <pubDate>Sat, 13 Jul 2013 23:00:00 GMT</pubDate>
            <description><![CDATA[This is an article about Hello World.]]></description>
            <content:encoded><![CDATA[Content of my item]]></content:encoded>
            <author>janedoe@example.com (Jane Doe)</author>
            <author>joesmith@example.com (Joe Smith)</author>
            <enclosure url=\\"https://example.com/hello-world.jpg\\">
            </enclosure>
        </item>
    </channel>
</rss>"
`;
